sdk/core/core-http/lib/policies/deserializationPolicy.ts

```diff
@@ -140,140 +140,131 @@ export function deserializeResponseBody(
   response: HttpOperationResponse
 ): Promise<HttpOperationResponse> {
   return parse(jsonContentTypes, xmlContentTypes, response).then((parsedResponse) => {
-    const shouldDeserialize: boolean = shouldDeserializeResponse(parsedResponse);
-    if (shouldDeserialize) {
-      const operationSpec: OperationSpec | undefined = parsedResponse.request.operationSpec;
-      if (operationSpec && operationSpec.responses) {
-        const statusCode: number = parsedResponse.status;
-
-        const expectedStatusCodes: string[] = Object.keys(operationSpec.responses);
-
-        const hasNoExpectedStatusCodes: boolean =
-          expectedStatusCodes.length === 0 ||
-          (expectedStatusCodes.length === 1 && expectedStatusCodes[0] === "default");
+    const shouldDeserialize = shouldDeserializeResponse(parsedResponse);
+    if (!shouldDeserialize) {
+      return parsedResponse;
+    }
 
-        const responseSpec: OperationResponse | undefined = getOperationResponse(parsedResponse);
+    const operationSpec = parsedResponse.request.operationSpec;
+    if (!operationSpec || !operationSpec.responses) {
+      return parsedResponse;
+    }
 
-        const isExpectedStatusCode: boolean = hasNoExpectedStatusCodes
-          ? 200 <= statusCode && statusCode < 300
-          : !!responseSpec;
-        if (!isExpectedStatusCode) {
-          const defaultResponseSpec: OperationResponse = operationSpec.responses.default;
-          if (defaultResponseSpec) {
-            const initialErrorMessage: string = isStreamOperation(operationSpec)
-              ? `Unexpected status code: ${statusCode}`
-              : (parsedResponse.bodyAsText as string);
+    const responseSpec = getOperationResponse(parsedResponse);
+    const statusCode = parsedResponse.status;
```

Comment:
nit: move `statusCode` right before it is used for better readability

Reply:
Removed the variable altogether, wasnt that useful